app/src/main/java/org/jboss/hal/client/configuration/subsystem/transactions/TransactionView.java

```diff
@@ -0,0 +1,199 @@
+        attributesForm.addFormValidation(formItems -> {
+
+            final FormItem<Boolean> journalStoreEnableAsyncIoItem = formItem(formItems, "journal-store-enable-async-io");
+            final FormItem<Boolean> useJournalStoreItem = formItem(formItems, "use-journal-store");
+
+            ValidationResult validationResult = ValidationResult.OK;
+
+            if (journalStoreEnableAsyncIoItem != null) {
+                final boolean journalStoreEnableAsyncIo = journalStoreEnableAsyncIoItem.getValue() != null && journalStoreEnableAsyncIoItem.getValue();
+                final boolean useJournalStore = useJournalStoreItem != null && useJournalStoreItem.getValue() != null && useJournalStoreItem.getValue();
+
+                if (journalStoreEnableAsyncIo && !useJournalStore) {
+                    useJournalStoreItem.showError("Journal store needs to be enabled before enabling asynchronous IO.");
+                    validationResult = invalid;
+                }
+            }
+            return validationResult;
+        });
+        
+        // --------------- form validation for the process attributes
+        processForm.addFormValidation(formItems -> {
+
+            ValidationResult validationResult = ValidationResult.OK;
+            FormItem<Boolean> uuidItem = formItem(formItems, PROCESS_ID_UUID);
+            FormItem<String> socketBindingItem = formItem(formItems, PROCESS_ID_SOCKET_BINDING);
+            FormItem<Number> socketMaxPortsItem = formItem(formItems, PROCESS_ID_SOCKET_MAX_PORTS);
+            if (uuidItem != null && socketBindingItem != null) {
+                boolean uuidGiven = uuidItem.getValue() != null && uuidItem.getValue();
+                String socketBinding = Strings.emptyToNull(socketBindingItem.getValue());
+
+                if ((uuidGiven && socketBinding != null) || (!uuidGiven && socketBinding == null)) {
+                    socketBindingItem.showError("Please set either UUID or socket binding");
+                    validationResult = ValidationResult.invalid("Validation error, see error messages below.");
+                }
+            }
+            if (socketBindingItem != null && socketMaxPortsItem != null) {
+                String socketBinding = Strings.emptyToNull(socketBindingItem.getValue());
+                Number socketMaxPorts = socketMaxPortsItem.getValue();
+
+                if (socketBinding == null && socketMaxPorts != null && socketMaxPortsItem.isModified()) {
+                    socketMaxPortsItem.showError("Can't be set if socket binding is not set");
+                    validationResult = invalid;
+                }
+            }
+            return validationResult;
+        });
+        
+        // set the process fields as not required, because uuid and socket-binding are mutually exclusive.
+        processForm.getFormItems().forEach(formItem -> formItem.setRequired(false));
+        
+        processForm.setSaveCallback(new Form.SaveCallback<ModelNode>() {
```

Comment:
I think it's better to move this code to the presenter and reference the presenter method in the MBUI xml (see below). Following the MVP paradigm we should try to have most of the logic in the presenter and only the UI related stuff in the view.